File: C++/Bryann Valderrama/jumpSearch.cpp
#include <iostream>
#include <cmath>

using namespace std;

int jumpSearch(int lista[], int x)
{
    int tamanio = sizeof(lista) / sizeof(lista[0]);
    int salto = (int)floor(sqrt(tamanio));
    int previo = 0;
    while (lista[min(salto, tamanio) - 1] < x)
    {
        previo = salto;
        salto = (int)salto + (int)floor(sqrt(tamanio));
        if (previo >= tamanio)
        {
            return -1;
        }
    }
    while (lista[previo] < x)
    {
        previo++;
        if (previo == min(salto, tamanio))
        {
            return -1;
        }
    }
    if (lista[previo] == x)
    {
        return previo;
    }
    return -1;
}

int main(int argc, char const *argv[])
{
    int lista[] = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144};
    int numeroabuscar;
    cin >> numeroabuscar;
    int resultado = jumpSearch(lista, numeroabuscar);
    if (resultado == -1)
    {
        cout << "No encontrado" << endl;
    }
    else
    {
        cout << "Encontrado en el indice " << resultado << endl;
    }
    return 0;
}

File: Codigos estudiantes por lenguaje/PY/Daniel Alayon/Algoritmos de busqueda/jumpSearch.py
import math


def jumpSearch(lista, x):
    tam = len(lista)
    salto = math.floor(math.sqrt(tam))
    previo = 0
    while lista[min(salto, tam) - 1] < x:
        previo = salto
        salto += math.floor(math.sqrt(tam))
        if previo >= tam:
            return -1
    while lista[previo] < x:
        previo += 1
        if previo >= tam:
            return -1
    while lista[previo] == x:
        return previo
    return -1


lista = [10, 20, 30, 40, 50, 60]
numeroBuscado = 10
resultado = jumpSearch(lista, numeroBuscado)
if resultado == -1:
    print("No encontrado")
else:
    print("Numero encopntrado en la posicion: ", resultado)
